### PR TITLE
Eliminate type checks for float args

### PR DESCRIPTION
**Pull request by billbudge**
Eliminate type checks and let the VM perform more relaxed ToInt32 or ToNumber conversion as laid out in the WebAssembly and JavaScript Specifications.

https://www.w3.org/TR/wasm-js-api-1/#towebassemblyvalue

This changes the behavior which only allows Booleans and Numbers. This change with ToNumber conversion would expand the allowed conversions with:

Undefined => NaN
Null => 0
String => StringToNumber(argument)   (converts to NaN if argument can't be parsed as number)
Object => ToNumber(ToPrimitive(argument))

For i32, NaN is converted to 0.

So broken inputs like Undefined would be silently converted to 0 in some cases.

We may therefore want to put this behind a flag.

### CODE REVIEW

File: src/embind/embind.js [hunk 1]
```diff
@@ -642,11 +642,8 @@ var LibraryEmbind = {
             return value;
         },
         'toWireType': function(destructors, value) {
-            // todo: Here we have an opportunity for -O3 level "unsafe" optimizations: we could
-            // avoid the following if() and assume value is of proper type.
-            if (typeof value !== "number" && typeof value !== "boolean") {
-                throw new TypeError('Cannot convert "' + _embind_repr(value) + '" to ' + this.name);
-            }
+            // The VM will perform JS to Wasm value conversion, according to the spec:
+            // https://www.w3.org/TR/wasm-js-api-1/#towebassemblyvalue
```
Comment (sbc100) [inline src/embind/embind.js:646]: How about we put the old check behind "#if ASSERTIONS`?

**Review by sbc100 — COMMENTED** [review]
Nice!

Also, welcome @billbudge!  I've invited you to join the emscripten project.

File: src/embind/embind.js (hunk 1 above)
Comment (billbudge) [reply src/embind/embind.js:651]: Thanks Sam, I put the old code back gated by ASSERTIONS.

**Review by sbc100 — APPROVED** [review]
lgtm.

Can you give the PR a more meaningful title?

Lets wait for @RReverser to lg before landing..

**Review by RReverser — APPROVED** [review]
Seems reasonable to me.

**Pull request merged**